app/controllers/spree/paypal_controller.rb

```diff
@@ -33,71 +32,135 @@ def express
+          flash[:success] = 'Payment created'
+          @order.reload
+          if params[:commit] == Spree.t(:submit_order)
+            respond_to do |format|
+              format.html { redirect_to success_order_path(@order) }
+              format.js { render js: "window.location.href = '" + success_order_path(@order) + "'" }
+            end
+          else
+            respond_to do |format|
+              format.html { redirect_to edit_order_path(@order) }
+              format.js { render js: 'window.location.reload();' }
+            end
+          end
+        end
+      rescue Spree::Core::GatewayError => e
+        # invoke_callbacks(:create, :fails)
+        @account_payment.destroy!
+        error_message = e.message == 'Internal Error' ? 'Unable to reach PayPal servers. Please contact help@getsweet.com for additional information.' : e.message
+        flash[:errors] = [error_message]
+        redirect_to edit_order_path(@order)
       end
     end
 
     def cancel
       flash[:notice] = Spree.t('flash.cancel', :scope => 'paypal')
-      order = current_order || raise(ActiveRecord::RecordNotFound)
-      redirect_to checkout_state_path(order.state, paypal_cancel_token: params[:token])
+      order = current_spree_user.company.purchase_orders.friendly.find(params[:order_id])
+      redirect_to edit_order_path(order)
     end
 
     private
 
     def line_item(item)
       {
-          :Name => item.product.name,
-          :Number => item.variant.sku,
-          :Quantity => item.quantity,
-          :Amount => {
-              :currencyID => item.order.currency,
-              :value => item.price
-          },
-          :ItemCategory => "Physical"
+        Name: item.product.name,
+        Number: item.variant.sku,
+        Quantity: item.quantity,
+        Amount: {
+          currencyID: item.order.currency,
+          value: item.price
+        },
+        ItemCategory: 'Physical'
       }
     end
 
-    def express_checkout_request_details order, items
-      { :SetExpressCheckoutRequestDetails => {
-          :InvoiceID => order.number,
-          :BuyerEmail => order.email,
-          :ReturnURL => confirm_paypal_url(:payment_method_id => params[:payment_method_id], :utm_nooverride => 1),
-          :CancelURL =>  cancel_paypal_url,
-          :SolutionType => payment_method.preferred_solution.present? ? payment_method.preferred_solution : "Mark",
-          :LandingPage => payment_method.preferred_landing_page.present? ? payment_method.preferred_landing_page : "Billing",
-          :cppheaderimage => payment_method.preferred_logourl.present? ? payment_method.preferred_logourl : "",
-          :NoShipping => 1,
-          :PaymentDetails => [payment_details(items)]
-      }}
+    def express_checkout_request_details(order, items)
+      { SetExpressCheckoutRequestDetails: {
+        InvoiceID: order.number + '_' + (order.payments.count + 1).to_s,
+        BuyerEmail: order.email,
```

Comment:
```suggestion
        BuyerEmail: order.valid_emails.first.to_s,
```
Assuming this is the email that gets populated on the paypal form.  This can contain multiple emails in Sweet, but can only be one for PayPal

Reply:
Fixed